### PR TITLE
Pass `gitUserEmailHash` up with new builds and use `localBuilds` filter in baseline calculations

### PR DESCRIPTION
**Pull request by tmeasday**
Telescoping on https://github.com/chromaui/chromatic-cli/pull/795

We want to mark who created each build[1] so we can filter out local builds created by anyone else apart from the current user.

As such we:

 - Send `gitUserEmaiHash ` as part of the `AnnounceBuild` mutation
 - Send `localBuilds: { localBuildEmailHash: gitUserEmaiHash || committerHash }` as part of all the queries we use for baseline calculations.

This depends on https://github.com/chromaui/chromatic/pull/7576

[1] The creator is the current git user for local builds and the committer otherwise.

### CODE REVIEW

File: node-src/tasks/gitInfo.ts
Comment (ghengeveld) [reply node-src/tasks/gitInfo.ts:69]: I don't think we should record the committer as the creator for non-local builds. It would make it harder to tell where the data originated (the commit or the git config) whilst not providing any benefit (we already record `committerEmail` so it would be duplicative).

Comment (ghengeveld) [reply node-src/tasks/gitInfo.ts:69]: We should consider hashing the email and recording `creatorId` or something like that instead, in order to avoid recording (even more) PII. We don't need the actual plaintext email, right? We can always hash the `committerEmail` as well if we want to cross-reference a CI build with a local build. Gravatar's [implementation](https://en.gravatar.com/site/implement/hash) (md5) seems reasonable.

**Review by ghengeveld — COMMENTED** [review]
There's some far-reaching consequences we should consider before merging this. Overall I'm not too keen on the implementation specifics right now.

File: node-src/tasks/gitInfo.ts
Comment (ghengeveld) [reply node-src/tasks/gitInfo.ts:69]: What happens if (for whatever reason) we can't retrieve the Git user email when running a local build? I think it should not be allowed to do that because we wouldn't be able to track local baselines.

File: node-src/types.ts
Comment (ghengeveld) [inline node-src/types.ts:150]: Within the context of the `git` object, this should be `userEmail`. Our APIs of course should still reference `creatorEmail` (or e.g. `creatorId`).

Comment (tmeasday) [reply node-src/types.ts:150]: I was using `userEmail` to refer to the `git config user.email`, where as `creatorEmail` is the slightly more complicated thing referenced in the comment above. I wanted to add the complicated thing to the `git` object because we need to pass it to various APIs along the way, so I didn't want to recalculate it each time (although I could).

File: node-src/tasks/gitInfo.ts
Comment (tmeasday) [reply node-src/tasks/gitInfo.ts:69]: > What's the purpose of recording unknown? Usually we do that only when retrieving data, not when recording it.

I guess we don't need to *record* `'unknown'`, as really the `creatorEmail` doesn't have any effect on non-local builds anyway (at the moment).

The purpose for it on non-local builds is in baseline selection (ie. queries we do). In these cases we want to pass *something* as `onlyLocalBuildsFrom` on CI builds if we can't calculate the commit's author. That will have the effect of filtering out *all* local builds, which I think is better than including all local builds.

We probably also need to show a large warning when this happens.

> Let's just always record the configured user email (preferably hashed), even for CI builds. We already track the committerEmail, so it would be duplicative anyway.

Given the above, we might consider not even storing a `creatorEmail` on CI builds? I could go either way.

> We should consider hashing the email and recording creatorId or something like that instead, in order to avoid recording (even more) PII. 

OK, sure, let's start hashing (the new) email fields.

> What happens if (for whatever reason) we can't retrieve the Git user email when running a local build? I think it should abort the build (throw an error) because we wouldn't be able to track local baselines. Falling back to unknown is likely to cause weird issues, but the above logic isn't doing that (for local builds) anyway.

I think it would throw an error but I'll verify that.

File: node-src/lib/emailHash.ts
Comment (ghengeveld) [inline node-src/lib/emailHash.ts:1]:
```suggestion
```
Doesn't seem appropriate to refer to private code from a public repo.

File: node-src/lib/localBuildsSpecifier.ts
Comment (ghengeveld) [inline node-src/lib/localBuildsSpecifier.ts:7]:
```suggestion
  // For global builds, we only want local builds from the committer (besides global builds)
```

File: node-src/tasks/initialize.ts
Comment (ghengeveld) [inline node-src/tasks/initialize.ts:67]:
```suggestion
        ...(gitUserEmail && { gitUserEmailHash: emailHash(gitUserEmail) }),
```
Otherwise we'll potentially get a hash of an empty string.

**Review by ghengeveld — APPROVED** [review]

File: node-src/tasks/gitInfo.ts
Comment (ghengeveld) [inline node-src/tasks/gitInfo.ts:70]: It seems this git command fails when there is no email configured, which is likely in a CI environment. 
```suggestion
    gitUserEmail: await getUserEmail().catch((e) => {
      ctx.log.debug('Failed to retrieve Git user email', e);
      return null;
    }),
```
I used this same pattern when [retrieving the uncommitted hash](https://github.com/chromaui/chromatic-cli/pull/798/files).

**Pull request merged**
